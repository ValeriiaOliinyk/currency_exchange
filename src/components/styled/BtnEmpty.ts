import styled from "styled-components";

interface BtnEmptyProps {
  type: string;
}

export const BtnEmpty = styled.section<BtnEmptyProps>`
  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 15px;
    height: 15px;
  }
  width: 20px;
  height: 20px;
  background-image: url(${`${window.location.origin}/images/starEmpty.svg`});
  background-repeat: no-repeat;
  background-size: cover;
`;
