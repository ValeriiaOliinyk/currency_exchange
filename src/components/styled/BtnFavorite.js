import styled from "styled-components";
import star from "../../images/star.svg";

export default styled.section`
  width: 20px;
  height: 20px;
  background-image: url(${star});
  background-repeat: no-repeat;
  background-size: cover;
`;
