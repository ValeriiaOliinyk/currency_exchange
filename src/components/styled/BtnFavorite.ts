import styled from "styled-components";

interface BtnFavoriteProps {
  type: string;
}

export const BtnFavorite = styled.section<BtnFavoriteProps>`
  width: 20px;
  height: 20px;
  background-image: url(${`${window.location.origin}/images/star.svg`});
  background-repeat: no-repeat;
  background-size: cover;
`;
