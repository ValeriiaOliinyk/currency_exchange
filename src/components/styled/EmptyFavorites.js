import styled from "styled-components";

export default styled.li`
  text-align: center;
  font-size: 20px;
  color: ${(props) => props.theme.colors.text};
  font-weight: 800;
`;
