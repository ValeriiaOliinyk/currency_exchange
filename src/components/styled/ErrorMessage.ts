import styled from "styled-components";

interface StyledProps {
  transparent?: boolean;
}

export const ErrorMessage = styled.div<StyledProps>`
  width: 100%;
  font-size: 18px;
  font-weight: 700;
  color: ${(props) =>
    props.transparent
      ? props.theme.colors.transparent
      : props.theme.colors.red};
`;
