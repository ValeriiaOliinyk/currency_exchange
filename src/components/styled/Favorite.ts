import styled from "styled-components";

export const Favorite = styled.div`
  margin: auto;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 22px;
  margin-top: 20px;
  width: 40px;
  height: 40px;
  background-image: url("./images/star.svg");
  background-repeat: no-repeat;
  background-size: cover;
  color: ${(props) => props.theme.colors.white};
`;
