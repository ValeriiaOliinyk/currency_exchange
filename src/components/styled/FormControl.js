import styled from "styled-components";
import { Form } from "react-bootstrap";

export default styled(Form.Control)`
  width: 75%;
`;
