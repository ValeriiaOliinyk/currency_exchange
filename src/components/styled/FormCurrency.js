import styled from "styled-components";
import { Form } from "react-bootstrap";

export default styled(Form)`
  margin: auto;
  width: 500px;
  height: 180px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.75);
  padding: 20px;
  border-radius: 20px;
`;
