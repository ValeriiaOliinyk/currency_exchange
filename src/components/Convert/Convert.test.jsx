import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { Container, Title } from "../../styled";

test("Container component works", () => {
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test("Container component works with Title component inside", () => {
//   const container = renderer.create(<Container />).toJSON();
//   ReactDOM.render(<Title />, container);
//   ReactDOM.unmountComponentAtNode(container);
// });
