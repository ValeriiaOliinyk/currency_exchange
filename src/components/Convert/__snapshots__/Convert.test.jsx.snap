// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component works 1`] = `
.c0 {
  width: 80%;
  margin: auto;
  margin-top: 100px;
}

<section
  className="c0"
/>
`;

exports[`Container component works with Title component inside 1`] = `
.c0 {
  width: 80%;
  margin: auto;
  margin-top: 100px;
}

<section
  className="c0"
/>
`;

exports[`it works 1`] = `
.c0 {
  width: 80%;
  margin: auto;
  margin-top: 100px;
}

<section
  className="c0"
/>
`;
